engine/crates/engine/src/headers.rs

```diff
@@ -24,7 +24,7 @@ impl RequestHeaders {
         )
     }
 
-    pub fn find(&self, expected_name: &str) -> Option<&str> {
+    pub fn find<'a>(&'a self, expected_name: &str) -> Option<&'a str> {
```

Comment:
Not necessary in this PR anymore, but it doesn't hurt to be more specific and extend the lifetime.